src/main/resources/assets/js/app/principal/PrincipalLoader.ts

```diff
@@ -13,4 +17,30 @@ export class PrincipalLoader
     protected createPreLoadRequest(principalKeys: PrincipalKey[]): GetPrincipalsByKeysRequest {
         return new GetPrincipalsByKeysRequest(principalKeys);
     }
+
+    // If the data is less than {MAX_TO_APPEND}, then just execute the "preLoadRequest".
+    // Otherwise, to improve performance, lib-admin-ui will not append items after {MAX_TO_APPEND}th one
+    // so there's no need to execute a request for those items to get the display name, therefore
+    // the data is a mix of the first {MAX_TO_APPEND} ones from the request and the remaining ones built directly
+    // from the keys in the searchString.
+    preData(searchString: string): Q.Promise<Principal[]> {
```

Comment:
Try to use method names that are self-explanatory rather than using short names and writing long comments explaining what the method does. Plus, "`preData`" doesn't sound like a verb. It should be "`preloadData`" or something.